compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -972,10 +972,19 @@ impl<'hir> Map<'hir> {
             }
             // Constants and Statics.
             Node::Item(Item {
-                kind:
-                    ItemKind::Const(ty, ..)
-                    | ItemKind::Static(ty, ..)
-                    | ItemKind::Impl(Impl { self_ty: ty, .. }),
+                kind: ItemKind::Impl(Impl { self_ty: ty, generics, .. }),
+                span: outer_span,
+                ..
+            }) => until_within(
+                *outer_span,
+                if generics.has_where_clause_predicates {
+                    generics.where_clause_span
+                } else {
+                    ty.span
+                },
```

Comment:
```suggestion
                generics.where_clause_span,
```
This should work in all cases. If there are no where clauses, we should have an empty span at the `ty`'s end.
Or do you want to skip the lone `where`?

Reply:
Lone `where` is fine to include, yeah.